src/mantine-core/src/components/Popper/PopperContainer/PopperContainer.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Portal } from '../../Portal';
+
+export interface PopperContainerProps {
+  /** PopperContainer children, for example, modal or popover */
+  children: React.ReactNode;
+
+  /** Root element z-index property */
+  zIndex?: number;
+
+  /** Root element className */
+  className?: string;
+
+  /** Whether to render the target element in a Portal */
+  withinPortal?: boolean;
+}
+
+export function PopperContainer({
+  children,
+  zIndex = 1,
+  className,
+  withinPortal,
+}: PopperContainerProps) {
+  if (withinPortal) {
+    return (
+      <Portal className={className} zIndex={zIndex}>
+        {children}
+      </Portal>
+    );
+  }
+
+  return (
+    <div className={className} style={{ position: 'relative', zIndex }}>
+      {children}
+    </div>
```

Comment:
I case of no-portal, I simply render what would be rendered by the `Portal` component, but outside of Portal itself.

In the end it gets very close to your `disabled` prop on `Portal` proposal I guess, but at least it is not exposed to the end-user that way.